### ISSUE TITLE
Wrong modeling of SearchRequest and its schema

### ISSUE THREAD

**Issue by jgomer2001** [GluuFederation/SCIM-Client#59]
[SearchRequest](https://github.com/GluuFederation/oxTrust/blob/version_3.1.1/scim/src/main/java/org/gluu/oxtrust/model/scim2/SearchRequest.java#L24) class does not model the `attributes` param correctly. According to spec this should be a (multi-valued) string list - section 3.4.3 RFC 7644. Currently, it's being used as a comma-separated string.

Additionally, the name used for serialization/deserialization is not correct (member is named `attributesArray`).  

The value `Constants.SEARCH_REQUEST_SCHEMA_ID` used to specify the schema is wrong as well (mis-spelled). Current value is `urn:ietf:params:scim:schemas:core:2.0:SearchRequest` while it should be `urn:ietf:params:scim:api:messages:2.0:SearchRequest`.

In other words, the POST search is using a payload that is not spec compliant.
